eth/beacon/state_machines/state_transitions.py

```diff
@@ -0,0 +1,36 @@
+from abc import (
+    ABC,
+    abstractmethod,
+)
+
+from eth.utils.datatypes import (
+    Configurable,
```

Comment:
@cburgdorf this has been problematic for typing right?  Thoughts on deprecating this pattern other than in cases where we require dynamic class construction (like the `Chain` API which needs to be able to be dynamically configured at runtime to support custom genesis configs)

Reply:
I have thought about it as well but I wasn't sure how much we still rely on the benefits of the pattern. E.g. is it needed for the sweet builder system that use to construct chains for testing?

Reply:
> Thoughts on deprecating this pattern other than in cases where we require dynamic class construction (like the Chain API which needs to be able to be dynamically configured at runtime to support custom genesis configs)

Do you mean `StateTransition` doesn't need to be `Configurable` since we can configure it in `StateMachine` level?

Reply:
The idea would be to get rid of  `Configurable` entirely. Here's some previous discussion about it https://github.com/ethereum/py-evm/issues/622